cmd/createcluster.go

```diff
@@ -386,43 +398,94 @@ func getValidators(dvs []tbls.TSS) ([]cluster.DistValidator, error) {
 	return vals, nil
 }
 
-// getOperators returns a list of operators from the provided secret keyshares.
-func getOperators(numNodes int, clusterDir string, insecureKeys bool, shareSets [][]*bls_sig.SecretKeyShare) ([]cluster.Operator, error) {
-	var peers []p2p.Peer
+// saveKeysToKeyManager saves validator keys to the provided keymanager addresses.
+func saveKeysToKeymanager(ctx context.Context, addrs []string, numNodes int, shareSets [][]*bls_sig.SecretKeyShare) error {
+	if len(addrs) != numNodes {
+		return errors.New("insufficient no of keymanager addresses", z.Int("expected", numNodes), z.Int("got", len(addrs)))
+	}
+
+	grp := new(errgroup.Group)
+	maxRetries := 5
 	for i := 0; i < numNodes; i++ {
-		peer, err := newPeer(clusterDir, i)
-		if err != nil {
-			return nil, err
+		var secrets []*bls_sig.SecretKey
+		for _, shares := range shareSets {
+			secret, err := tblsconv.ShareToSecret(shares[i])
+			if err != nil {
+				return err
+			}
+			secrets = append(secrets, secret)
 		}
 
-		peers = append(peers, peer)
+		retries := maxRetries
+		addr := addrs[i]
+		grp.Go(func() error {
+			for retries > 0 {
+				err := keystore.PostKeysToKeymanager(ctx, addr, secrets)
+				if err == nil {
+					log.Debug(ctx, "Pushed keys to keymanager", z.Str("addr", addr))
+					return nil
+				}
+
+				retries--
```

Comment:
this variable isn't being a lock but it is being updated concurrently. Suggest removing retries. Rather just check if the keymanagers are accessible at the start, if not fail, otherwise just write and fail.